data-syncs/supported-data-sync-sources/soap-1.2-web-CDC.md

```diff
@@ -0,0 +1,97 @@
+| Variables   | **Optional.** Review our documentation on [Variables here ](../building-data-syncs/advanced-settings/variables.md)for more information about this field.                                         |           |
+| Permissions | Data syncs are role based access systems where you can give specific groups read, write, execute, and/or all of the above with admin access. **Inputting at least an Admin Group is mandatory.** |           |
+
+<figure><img src="../../.gitbook/assets/image (166).png" alt=""><figcaption><p>Image 1: The Info Tab</p></figcaption></figure>
+
+## Source tab
+
+The following table outlines the mandatory and optional parameters you will find
+on the Source tab _(Image 2)._
+
+{% tabs %} {% tab title="Source Details" %} The following parameters will help
+to define your data sync source and how it functions.
+
+| Parameter           | Description                                                                                                          | Example                                        |
+| ------------------- | -------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------- |
+| Source              | Mandatory. Select your source from the drop down menu.                                                               | SOAP 1.2 Web Service                           |
+| authType            | Mandatory. Select the type of authentication you wish to use in this sync: None, WSSE, Basic.                        | Basic                                          |
+| Use Password Digest | Use only with WSSE authType and Password Type as **PasswordDigest**. Otherwise, leave unchecked.                     |                                                |
+| Request Timeout     | Mandatory. Set a timeout in milliseconds. No maximum value. Minimum greater than 0. Default is 100 milliseconds.     | 2000                                           |
+| Endpoint            | Mandatory. Contains your SOAP 1.2 Web Service API endpoint.                                                          | `dataaccess`                                     |
+| Has Mtom Response   | Required to be true if SOAP API response contains an attachment outside the message.                                 |                                                |
+| Record Xpath        | Mandatory. The Xpath to select records to extract from the SOAP response. Starts with ‘//’ followed by the tag name. |                                                |
+| Envelope Namespace  | Namespace prefix for SOAP request elements.                                                                          | "foo"                                          |
```

Comment:
This example is not helpful.

Given the following semantics, see if you can find a better way to explain what this does.

Imagine your SOAP **response** is like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope">
    <soap:Body>
        <m:ListOfLanguagesByNameResponse xmlns:m="http://www.oorsprong.org/websamples.countryinfo">
            <m:ListOfLanguagesByNameResult>
                <m:tLanguage>
                    <m:sISOCode>abk</m:sISOCode>
                    <m:sName>Abkhazian</m:sName>
                </m:tLanguage>
                <m:tLanguage>
                    <m:sISOCode>ace</m:sISOCode>
                    <m:sName>Achinese</m:sName>
                </m:tLanguage>
            </m:ListOfLanguagesByNameResult>
        </m:ListOfLanguagesByNameResponse>
    </soap:Body>
</soap:Envelope>
```

Then the xpath for the records would be //m:tLanguage   (note that's the repeating element which contains the records)

Then for each prefix x:y e.g. `soap:*` or `m:*` (i.e. the namespace), you need to tell SOAP what the expected structure of that element is, and the namespace definition resides at a particular url.

in this case namespace name **m** is at namespace value (url) **http://www.oorsprong.org/websamples.countryinfo**
and **soap** is at **http://www.w3.org/2003/05/soap-envelope**.

One confusing thing is that to send our **request**, we're actually using the **soap12** namespace e.g.

```xml
<soap12:Body xmlns:soap12="http://www.w.org/2003/05/soap-envelope">
<ListOf LanguagesByName xmlns="http://www.oorsprong.org/websamples.countryinfo" />
</soap12:Body>
```

So (my understanding is that) the poor user has to always include the `soap12` => `http://www.w.org/2003/05/soap-envelope` namespace mapping regardless, while not having a clue that that's expected...

hopefully we fix this in the future...

Reply:
Noted. We'll go over this in our meeting today.